examples/cpp/main.cpp

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (2019) Cobalt Speech and Language, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cubic_client.h"
+#include "cubic_exception.h"
+#include "demo_config.h"
+#include "recorder.h"
+#include "timer.h"
+
+#include <atomic>
+#include <iostream>
+#include <regex>
+#include <string>
+#include <thread>
+
+
+// Wait for the Enter key to be pressed
+void waitForEnter()
+{
+    // This is a somewhat simplistic way to detect if the enter key was
+    // pressed, but it is sufficient for demo purposes.
+    std::string line_input;
+    std::getline(std::cin, line_input);
+}
+
+// Get the transcript from the response
+std::string extractTranscript(const cobaltspeech::cubic::RecognitionResponse &response)
+{
+    std::string transcript;
+    for(int i = 0; i < response.results_size(); i++)
+    {
+        cobaltspeech::cubic::RecognitionResult result = response.results(i);
+
+        // Ignore partial results
+        if(result.is_partial())
+            continue;
+
+        // If we don't have any alternatives, there isn't much for us
+        // to do right now.
+        if(result.alternatives_size() == 0)
+            continue;
+
+        // Just use the first alternative for now.
+        std::string raw_transcript = result.alternatives(0).transcript();
```

Comment:
Should we call this `raw_transcript` if we are calling `transcript()`.  Isn't there a `raw_transcript()` call available (not sure if that's the right function call.)

Reply:
You're right! I forgot that was added to the API. I will change the variable name.